boards/xtensa/intel_s1000_crb/intel_s1000_crb.dts

```diff
@@ -35,11 +36,53 @@
 
 &spi0 {
 	status = "ok";
+	#address-cells = <1>;
+	#size-cells = <1>;
 
-	mx25um512@0 {
+	flash0: flash@0 {
 		compatible = "spi,flash";
 		label = "MX25UM512";
-		reg = <0x00>;
+		reg = <0x00 0x400000>;
```

Comment:
This should just be reg = <0>;

Reply:
I get a parse error on XCC if i keep this as 0. In any case, i need to specify the size of the flash. So, i need to mention 0x400000 in some manner.

Reply:
right now dts/bindings/slave/spi,flash.yaml has a size property for this.

Reply:
Why can't the size be discovered via SFDP and used? I don't see a reason to limit the size to 4MB in device tree. The size should be discovered and the usage limit of 4MB should be described through the partitions definition as is being done in this PR.